### PR TITLE
Add functions

### PR DESCRIPTION
**Pull request by alexmedinasf**
Created webpack
Created Html java and css
Styled using css
Gave all functionalities with javascript

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules/
+package-lock.json
+hint-report/
```
Comment (franclobo) [inline .gitignore:3]: - [ ] Please 🙏🏽 just the node modules folder must be in the gitignore file. Take out the other files.

File: dist/index.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<!doctype html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>Webpack Set-Up</title><script src="https://kit.fontawesome.com/64d25ab310.js" crossorigin="anonymous"></script><script defer="defer" src="./main.js"></script></head><body><section id="main-container"><div id="main-c-head"><h3>Today's To Do</h3><i class="fa-solid fa-arrows-rotate"></i></div><input placeholder="Add to your list" id="to-do-input"><article id="to-do-list"></article><p id="clear-completed" class="clear-completed">Clear all completed</p></section></body></html>
```
Comment (franclobo) [inline dist/index.html:1]: - [ ] Kindly put the document in order, line by line. It is easier to review and use the inline comments in the corresponding place. 📑 👀

File: dist/main.js [hunk 3]
```diff
@@ -0,0 +1 @@
...assList.contains("list-items-c")){e.target.classList.toggle("new-background"),e.target.firstElementChild.nextElementSibling.classList.toggle("item-removed"),e.target.firstElementChild.nextElementSibling.nextElementSibling.classList.toggle("remove"),e.target.firstElementChild.lastElementChild.contentEditable="true";const t=e.target.firstElementChild.nextElementSibling.getAttribute("id");e.target.firstElementChild.lastElementChild.addEventListener("keypress",(e=>{if("Enter"===e.key){e.target.contentEditable="false";const n=e.target.innerHTML;e.target.parentElement.parentElement.classList.toggle("new-background"),e.target.parentElement.nextElementSibling.classList.toggle("item-removed"),e.target.parentElement.nextElementSibling.nextElementSibling.classList.toggle("remove"),b.updateDescription(n,t)}}))}if(!0===e.target.classList.contains("to-do-check")){e.target.parentElement.classList.toggle("line-through"),e.target.classList.toggle("false"),e.target.classList.toggle("true");const t=e.target.parentElement.nextElementSibling.getAttribute("id");E.updateStorageCheck(e.target,t)}})),L.addEventListener("click",(()=>{E.clearAllCompleted(),b.updateIndex(),window.location.reload()}))})()})();
```
Comment (franclobo) [inline dist/main.js:1]: - [ ] Kindly put the document in order, line by line. It is easier to review and use the inline comments in the corresponding place. 📑 👀

File: package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    "eslint": "^7.32.0",
+    "eslint-config-airbnb-base": "^14.2.1",
+    "eslint-plugin-import": "^2.26.0",
+    "html-webpack-plugin": "^5.5.0",
+    "style-loader": "^3.3.1",
+    "webpack": "^5.72.1",
+    "webpack-cli": "^4.9.2",
+    "webpack-dev-server": "^4.9.1"
+  }
+}
```
Comment (mckent05) [inline package.json:33]: - You seem to have installed webpack wrongly, I have some errors starting your project on my machine 👇 👇 . I suggest you add this `"start": "webpack serve --open",` to your scripts in the package.json file. Please try to fix this

Error:
webpack

File: src/check-status.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  static clearAllCompleted() {
+    const items = LocalStorage.getItemStorage();
+
+    for (let i = 0; i <= items.length; i++) {
+      items.forEach((item, index) => {
+        if (item.completed === true) {
+          items.splice(index, 1);
+        }
+      });
+    }
```
Comment (franclobo) [inline src/check-status.js:33]: - [ ] Please 🙏🏽 use the Array.filter method in the `Clear All Completed` button. It is a requirement:

> Implement a function for the "Clear all completed" button (use filter() method).

**Review by ishimwezachee — APPROVED** [review]
Hi @alexmedinasf , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Linters are passing ✅
Project is running successfully ✅

## Optional suggestions

- [ ]  In your PR description, you've confused java and javascript, just fix that because java is total different from javascript programing language. 

- [ ]  Your project is working fine, but I can see that you've implemented some features that are supposed to be done in the future milestone. You should implement the future requirements only if you've been approved for the current milestone by the code reviewer.  Don't repeat that because as a reviewer there will be some requirements that I will miss in your project because you've jumped to the future milestone. 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 6]
```diff
@@ -1 +1,30 @@
...
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Microverse for the opportunity
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (mckent05) [inline README.md:30]: - In order to make your repo look more professional, please include steps on how a beginner can run your project locally 👍 👍

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @alexmedinasf,

# Required Changes ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️
- Correct Github flow ✔️
- Descriptive PR ✔️
- Professional README file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [ ] I couldn't open your project to review the local storage in the browser, please fix that error:

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @alexmedinasf,

# STATUS: CHANGES REQUIRED ♻️ ♻️  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- Your project is working correctly 👏 
- All linter checks are passing ✔️ ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahmetbozaci** [comment]
⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ 
Hello, @alexmedinasf if the next code reviewer will approve it please open a new branch for the next milestone. You use the same branch for all milestones until now which is wrong. You should have different branches for the different milestones.

**Pull request merged**